### PR TITLE
Added a simple way to configure your own reporter to store coverage data

### PR DESCRIPTION
**Pull request by sunblaze**
Added a very simple way to configure our own reporter. Coverband's report generation will not work with this as it still uses the redis interface. But we'll worry about that once we actually start getting events in Kafka.

@eapache

### CODE REVIEW

File: lib/coverband/base.rb [hunk 1]
```diff
@@ -42,6 +42,7 @@ def reset_instance
       @ignore_patterns = Coverband.configuration.ignore
       @sample_percentage = Coverband.configuration.percentage
       @reporter = Coverband::RedisStore.new(Coverband.configuration.redis) if Coverband.configuration.redis
+      @reporter = Coverband.configuration.reporter if Coverband.configuration.reporter
```
Comment (eapache) [inline lib/coverband/base.rb]: I'd rather merge these two lines, so `if reporter set it, else if redis, set it to the redis-store`

**Comment by sunblaze** [comment]
Merged into one line the `@reporter` setting

**Comment by eapache** [comment]
Ah, I didn't really mean one line, I just meant one logical switch, but this is fine I think :)

**Comment by sunblaze** [comment]
ah, I know what you mean now

**Pull request merged**